core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1715,7 +1715,7 @@ class SparkContext(config: SparkConf) extends Logging {
   private[spark] def getRDDStorageInfo(filter: RDD[_] => Boolean): Array[RDDInfo] = {
```

Comment:
There's a single call to this method outside of tests, in `RDD.toDebugString`. That to me makes it another candidate to go away and be replaced with information from the `AppStatusStore`. Then maybe you can remove more code from `StorageStatus`.

Have you taken a look at that?

Reply:
I have found something and I am not sure whether is it a bug. 
Using rddStorageInfo.numCachedPartitions gives back a different value then the old storage util computed/updated rddInfo.numCachedPartitions.

This is why I changed the assert in org.apache.spark.repl.SingletonReplSuite at "replicating blocks of object with class defined in repl".

Reply:
If those values differ then it's probably a bug in the new code. Or maybe a bug in the old code, although that's less likely. It would be good to investigate why they differ.

Reply:
The old code considered the replication factor. I have created a separate Jira issue: https://issues.apache.org/jira/browse/SPARK-23394.